compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7301,4 +7309,102 @@ private STNode parseErrorKeyWord() {
             return sol.recoveredNode;
         }
     }
+
+    /**
+     * Parse let expression.
+     * <p>
+     * <code>
+     * let-expr := let let-var-decl [, let-var-decl]* in expression
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseLetExpression() {
+        STNode letKeyword = parseLetKeyword();
+        STNode letVarDeclarations = parseLetVarDeclarations();
+        STNode inKeyword = parseInKeyword();
+        STNode expression = parseExpression();
+        return STNodeFactory.createLetExpressionNode(letKeyword, letVarDeclarations, inKeyword, expression);
+    }
+
+    /**
+     * Parse let-keyword.
+     *
+     * @return Let-keyword node
+     */
+    private STNode parseLetKeyword() {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.LET_KEYWORD) {
+            return consume();
+        } else {
+            Solution sol = recover(token, ParserRuleContext.LET_KEYWORD);
+            return sol.recoveredNode;
+        }
+    }
+
+    /**
+     * Parse let variable declarations.
+     * <p>
+     * <code>let-var-decl-list := let-var-decl [, let-var-decl]*</code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseLetVarDeclarations() {
+        startContext(ParserRuleContext.LET_VAR_DECL);
+        List<STNode> varDecls = new ArrayList<>();
+        STToken nextToken = peek();
+
+        // Make sure at least one let variable declaration is present
+        if (isEndOfLetVarDeclarations(nextToken.kind)) {
+            this.errorHandler.reportMissingTokenError("missing let variable declaration");
+            return STNodeFactory.createMissingToken(SyntaxKind.LET_VAR_DECL);
```

Comment:
Should we return an empty list here?

Reply:
yeah. the context should be ended. :) 
is it ok to return an empty list? there should be at least one let-var-decl.
Following had done in the `parseListeners()` method.
```if (isEndOfExpressionsList(nextToken.kind)) {
            endContext();
            this.errorHandler.reportMissingTokenError("missing expression");
            return STNodeFactory.createMissingToken(SyntaxKind.IDENTIFIER_TOKEN);
        }

Reply:
Yeah, since we log the errors, thats should be fine.